doc/drawingElement.md

```diff
@@ -0,0 +1,1186 @@
+</pl-drawing>
+```
+<img src="elements/pl-drawing.png" width=70%>
+
+#### Customizations
+
+Attribute | Type | Default | Description
+--- | --- | --- | ---
+`preview` | boolean | false | `preview = true` is used when canvas is used for display only. `preview = false` expects objects to be placed in the canvas for grading.
+`answers-name` | string | — | Variable name to store data in. This variable is required when `preview = false`.
+`width` | integer | 580 | Horizontal width of the canvas.
+`height` | integer | 320 | Vertical height of the canvas.
+`grid-size` | integer | 20 | Size of the square grid for the canvas background. If `grid-size = 0`, then the background is white.
+`snap-to-grid` | boolen | true | Objects placed in the canvas snap to closest grid point.
+`correct-answer` | string | special | Correct answer for grading. Defaults to `data["correct-answers"][answers-name]`.
+`tol` | float | `0.5*grid-size` | Tolerance to check the position of objects. The error is calculated as the absolute difference between expected position and submitted one.
+`angle-tol` | float | 10 | Tolerance to check angles (in degrees). The error is calculated as the absolute difference between expected angle and submitted one.
+`show-tolerance-hint` | boolean | true | Show tolerance hint under the canvas. The default is `true` when `preview = false`
+`tolerance-hint` | text | "The expected tolerance is 1/2 square grid." | Hint explaining tolerance used when grading objects.
+
+
+## `pl-drawing-initial` element
+
+A `pl-drawing-initial` will group objects that will be displayed in the canvas defined by `pl-drawing`.
+
+For example, `pl-drawing-initial` can be used to create figures that are displayed as part of the question body:
+
+```html
+<pl-drawing preview=true grid-size="0">
+    <pl-drawing-initial>
+        <!-- create the image here -->
+    </pl-drawing-initial>
+</pl-drawing>
+```
+
+<img src="elements/pl-I.png" width=30%>
+
+And `pl-drawing-initial` can also be used to display initial objects in a canvas that will be used for grading. Objects inside `pl-drawing-initial` are not graded. Objects inside `pl-drawing-answer` are  graded.
+
+```html
+<pl-drawing answers-name="add-objects">
+
+    <pl-drawing-initial>
+    <!-- objects that define the correct answer are placed here -->
+    </pl-drawing-initial>
+    <pl-drawing-answer>
+      <!-- objects that define the correct answer are placed here -->
+    </pl-drawing-answer>
+
+</pl-drawing>
+```
+
+The child element `pl-drawing-answer` will be explained in the Grading section.
+
+# Elements to create drawing objects
+
+## `pl-coordinates` element
+
+A `pl-coordinates` element adds a 2D system of coordinates.
+
+#### Sample Element
+
+```html
+<pl-drawing preview=true width="200" height="200">
+    <pl-drawing-initial>
+        <pl-coordinates x1="60" y1="80" label="0" angle="30"></pl-coordinates>
+    </pl-drawing-initial>
+</pl-drawing>
+```
+<img src="elements/pl-coordinates.png" width=50%>
```

Comment:
I think the overarching doc style primarily uses Markdown to include images instead of the HTML `<img>` tag.

```md
![Example of `pl-coordinates`](elements/pl-coordinates.png)
```

 The downside is the [`mkdocs` build](https://github.com/mkdocs/mkdocs/issues/1385) doesn't allow for attributes as it uses [`Python-Markdown`](https://python-markdown.github.io/), which adheres to the [original Markdown spec](https://daringfireball.net/projects/markdown/syntax#img). What is the need to resizing the image? Could it be accomplished via CSS?

Reply:
From Slack... Turns out there is an option:

```md
![alt](path/to/image.png){: width="30%" }
```
c.f. https://python-markdown.github.io/extensions/attr_list/#inline

We might need to add to the `mkdocs.yml` config:

```yaml
markdown_extensions:
  - markdown.extensions.attr_list: true
```

c.f.
https://python-markdown.github.io/extensions/
https://www.mkdocs.org/user-guide/configuration/#formatting-options

Reply:
As discussed via Slack, we will work on these changes at a later time. Thanks for the input!